packages/dds/merge-tree/Obliterate.md

```diff
@@ -0,0 +1,276 @@
+    // happened, no need to continue.
+    return currentMin > smallestSeqObliterateOp;
+  }
+```
+
+We don't need to worry about the analogous problem of extending the excursion as a result of segments between the insert location and a local obliterate
+because any such segments would have also been marked as locally obliterated when they were inserted into the merge tree.
+In the sample code written for the remote segment, this will also necessitate `markSegmentObliterated` to tolerate marking segments with local obliteration info.
+
+Much of the same logic that goes into conflicting local + remote removal will need to be applied for obliterate.
+Nothing stands out as a conceptual issue or hurdle in this realm, though. Just tricky conditionals.
+
+Once the op is acked, the behavior in the [Remote perspective](#remote-perspective) section suffices for any further concurrent segments.
+
+### Other aspects
+
+TODO: Fill out this section in detail and with as much care as was taken above. For now (draft PR for early feedback), here is my current thinking on the areas I called out:
+
+- Any changes to direct fields of tree nodes themselves (either new data or changes to bookkeeping of existing data)
+> I think `{ obliteratedSeq?: number, localObliteratedSeq?: number }` suffices here. We'll of course need to make sure it's updated where appropriate (segment splits, checks for segment merges, etc.)
+- How the feature interacts with an increasing collab window and zamboni
+> I don't expect many conceptual issues here given the parallels with removedSeq, but need to do a code audit.
+- Impact on the partial lengths scheme
+> This does deserve its own section. Current partial lengths bookkeeping works for removal by adding various adjustment entries based on segments' `removedSeq` and overlapping removes fields. We need to update the bookkeeping on initial obliterate handling as well as on immediate-obliterate-on-inserting-walk cases.
+- Bookkeeping and handling of overlapping removals (note some may be obliterates and some may not be)
+> This was partially covered above. Not sure what else I want to add, but leaving it for now.
+- Reconnection
+> We might need to do fixup on the local state of the merge tree on reconnection by clearing rebased obliterate ops. I don't think the problem is intractible, and can always be done unperformantly in V1 since reconnection of an obliterate should be relatively uncommon.
+- Snapshotting impact
+> We'll need to ensure that obliterated info ends up on segments within the collab window. Otherwise I don't see issues here.
+
+## Endpoint Behavior
+
+One important consideration is what happens near the endpoints of the removed range.
+There are two general possibilities: either the obliterate expands to include segments inserted
+adjacent to the endpoint, or it does't.
+
+```typescript
+
+/** @sealed */
+interface IEndpointMergeStrategy {
+  // TODO
+}
+
+/** @sealed */
+interface IRangeMergeStrategy {
```

Comment:
i'd really like to avoid a strategy like this if possible, and just use strict positioning, (start, end], specifically if the pos of an incoming op with a refseq less than the obliterate is `pos >= start, and pos <end`. in general, i think it's probably the case the obliterate should win. This is basically the same as remove.

additionally, if we find that a default policy like this doesn't work, we can also add something like this later if it is really necessary

Reply:
Fair enough. I didn't comment on which policy we wanted here, because I don't think it impacts the design much. I left a note saying we could expose this in the future if we wanted to, but for now we should pick one which is in line with our approach for `insert`.